site/en/blog/astro-view-transitions/index.md

```diff
@@ -0,0 +1,164 @@
+
+## Early explorations with view transitions
+
+Chrome's journey with the View Transitions API was not a solitary one. Feedback and collaboration from developers, framework authors, and the CSS Working Group all played a crucial role in shaping the feature over the course of several years.
+
+Developer experimentation played a critical role early on. The earliest versions of the API didn't support CSS animations, and transitions were restricted to a few restrictive presets. Early feedback made it clear that developers wanted transitions to be fully customizable and expressive. This back-and-forth collaboration ensured that the API was designed with a balance of good defaults, extensibility, and customization.
+
+Astro was one of those frameworks that leaned into View Transitions early. The rest of this article is from the point of view of the Astro team, and shares their experience finding, adopting, and bringing View Transitions front-and-center to the web development experience.
+
+## Why Astro bet on view transitions
+
+Astro had been looking for something like View Transitions for a long time. Astro is a JavaScript web framework that server-renders your UI components to lightweight HTML for faster page load performance. Astro intentionally moves as much work off the client device as possible. Astro websites use native browser page navigation, where other web frameworks might hijack browser navigation with JavaScript client-side routing instead.
+
+This tradeoff created a challenge for the Astro team: How could Astro animate page transitions and provide app-like UI persistence across pages without adopting client-side routing? 
+
+<figure>
+{% Img src="image/1L2RBhCLSnXjCnSlevaDjy3vba73/SIBoP8VgOkV2bBWTWmwE.png", alt="ALT_TEXT_HERE", width="800", height="678" %}
+  <figcaption>
+    Challenges of animating page transitions with Astro
+  </figcaption>
+</figure>
+
+
+View Transitions were the answer. With View Transitions, Astro could provide the same routing features that JavaScript-heavy SPA web frameworks offered, but without the performance and complexity overhead of client-side routing. Other frameworks approached View Transitions as an optional implementation detail, but for Astro it was something much bigger.
+
+<figure>
+{% Img src="image/1L2RBhCLSnXjCnSlevaDjy3vba73/cG7Evh8rZRMB2q0wObHx.png", alt="ALT_TEXT_HERE", width="800", height="455" %}
+  <figcaption>
+    View Transitions as the answer
+  </figcaption>
+</figure>
+
+
+The next question the Astro team had to answer was, “how?” The team looked to projects like Turbo, Swup, and Livewire for inspiration and even prototyped a few similar approaches inside of Astro. 
+
+One of the earliest attempts involved a full-page “smart transition” that would automatically compare every new page of HTML to the current HTML and then attempt to replace every element at once using JavaScript. It was clever, but error-prone.
+
+```javascript
```

Comment:
all the code is indented, can you remove the white space at the beginning of lines?

Reply:
Ack. we will do this shortly.

Reply:
Can you check now.